data_rentgen/db/scripts/clean_partitions.py

```diff
@@ -0,0 +1,204 @@
+from enum import Enum
+from typing import Literal
+
+from dateutil.parser import isoparse
+from dateutil.relativedelta import relativedelta
+from sqlalchemy import text
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from data_rentgen.db.factory import create_session_factory
+from data_rentgen.db.settings import DatabaseSettings
+from data_rentgen.logging.settings import LoggingSettings
+from data_rentgen.logging.setup_logging import setup_logging
+
+logger = logging.getLogger(__name__)
+
+PARTITION_GRANULARITY_PATERN = re.compile(r"(\w+)_y(\d{4})(?:_m(\d{1,2})(?:_d(\d{1,2}))?)?")
+
+
+granularity_output_format = {
+    "year": "y%Y",
+    "month": "y%Y_m%m",
+    "day": "y%Y_m%m_d%d",
+}
+
+
+class Command(str, Enum):
+    DRY_RUN = "dry_run"
+    DETATCH_PARTITIONS = "detach_partitions"
+    REMOVE_DATA = "remove_data"
+    TRUNCATE = "truncate"
+
+
+def get_parser() -> ArgumentParser:
+    parser = ArgumentParser(
+        usage="python3 -m data_rentgen.db.scripts.clean_partitions command truncate --keep-after 2025-01-01",
+        description="Truncate or detach partitions, before provided date.",
+    )
+    parser.add_argument(
+        "command",
+        type=Command,
+        choices=[item.value for item in Command],
+        default=Command.DRY_RUN.value,
+        nargs="?",
+        help="Operation mode.",
+    )
+    parser.add_argument(
+        "--keep-after",
+        type=isoparse,
+        default=(datetime.now(tz=UTC) - relativedelta(years=1)),
+        nargs="?",
+        help="Partitions with data before this date will be considered for cleanup",
+    )
+    return parser
+
+
+@dataclass()
+class TabelPartitions:
+    partitions: list[date] = field(default_factory=list)
+    granularity: Literal["year", "month", "day"] = field(default="year")
+
+
+async def get_partitioned_tables(session: AsyncSession) -> dict[str, TabelPartitions] | None:
+    tables: dict[str, TabelPartitions] = defaultdict(TabelPartitions)
+    query = text(
+        "select c.relname as table_name from pg_class c where c.relispartition = True and c.relkind = 'r' order by c.relname",  # noqa: E501
+    )
+    result = await session.execute(query)
+    table_names = result.all()
+    if not table_names:
+        logger.info("There is no partitioned tables")
+        return tables
+    # parse tabel_names
+    for (tabel_name,) in table_names:
+        granularity: Literal["year", "month", "day"] = "year"
+        match = re.search(PARTITION_GRANULARITY_PATERN, tabel_name)
+        (name, year, month, day) = match.groups()  # type: ignore[union-attr]
+        year = int(year)
+        if month:
+            month = int(month)
+            granularity = "month"
+        else:
+            month = 1
+        if day:
+            day = int(day)
+            granularity = "day"
+        else:
+            day = 1
+        tables[name].partitions.append(date(year, month, day))
+        tables[name].granularity = granularity
```

Comment:
What if there are partitions with different granularities?

My proposal is to have class:
```python
@dataclass
class TablePartition
  name: str
  date: date
  granularity: "day" | "month" | "year"

  @property
  def qualified_name(self):
    return self.name + self.date.strptime(granularity_output_format[self.granularity])
```

And replace `dict[table_name, TablePartitions]` with `dict[table_name, list[TablePartition]]`